### ISSUE TITLE
Snapshot sizes support?

### ISSUE THREAD

**Issue by biji** [openSUSE/snapper#703]
Does snapper support showing snapshot sizes for `btrfs`? 
I've read http://snapper.io/2018/10/04/used-space.html , still don't understand what to put in config: `QGROUP`

Thanks

**Comment by aschnell** [comment]
From what I see that should work. Additionally, if querying the sizes fails the column is simply not shows, but there should be a response anyway.

What distributions are you using? And suspicious is dmesg, journtalctl or /var/log/snapper.log?

**Comment by biji** [comment]
while doing `snapper list`, high cpu
![image](https://user-images.githubusercontent.com/665559/160375973-a0a8c3dc-57f7-4572-bf52-e5ea59403634.png)

dmesg:
```
[18387.092068] BTRFS warning (device nvme0n1p7): qgroup rescan is already in progress
[18434.101623] BTRFS warning (device nvme0n1p7): qgroup rescan is already in progress
[18440.735865] BTRFS warning (device nvme0n1p7): qgroup rescan is already in progress
[18476.714101] BTRFS info (device nvme0n1p7): qgroup scan completed (inconsistency flag cleared)
```

I'm using fedora 35

I think snapper trigger rescan again , everytime i run `snapper list`

**Comment by aschnell** [comment]
Yes, a rescan is triggered very time - it is so far the only way to get the values right.

**Comment by biji** [comment]
Is it necessary after `qgroup scan completed (inconsistency flag cleared)`?

**Comment by aschnell** [comment]
Right after a qgroup scan is should not be needed but snapper does not know that a qgroup scan was just done.

Simply looking for the inconsistency flag did not work back when I implemented the display of those values in snapper.

Anyway, if the qgroup rescan takes that long it looks like a kernel or filesystem problem to me (unless the filesystem is very big, has lots of snapshots, ...).

**Comment by pa-sowa** [comment]
On my system with 938 GiB already allocated, qgroup scan takes 13 minutes, this makes `snapper list` unusable.

**Comment by aschnell** [comment]
If it is too slow you can use the option ```--disable-used-space```.

**Comment by pa-sowa** [comment]
> i think after `qgroup scan completed` , qgroup will be updated in realtime

The problem is snapper always triggers a rescan. So every time I invoke `snapper` list I have to wait 13 minutes.

**Comment by patphobos** [comment]
Hi, it seems we're encountering the same issue. The `snapper list` command, via the dbus service, appears to trigger BTRFS_IOC_QUOTA_RESCAN every time it runs, even if no new snapshots have been created.

The filesystem isn't particularly large: 17 TB, with 234,954 files and 12,676 folders. Currently, there are only 5 snapshots, and the refresh time is relatively long, about 15 minutes. However, I assume this behavior is normal and expected.

It's unfortunate that I'm considering repackaging snapper with quota support disabled so that the client isn't disrupted by quota refreshes. Instead, I'd monitor snapshot usage using `btrfs qgroup show`...

**Comment by ilkka-ollakka** [comment]
> Yes, a rescan is triggered very time - it is so far the only way to get the values right.

Is this still the case? For me it seems that btrfs fi sync path is only needed after quota is enabled to get right values with kernel 6.x.
